src/main/webapp/panelStyle.js

```diff
@@ -0,0 +1,168 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable no-unused-vars */
+
+// Script for handling the behaviour of the panels' layout
+
+// Add appropriate event listeners to all toggle icons
+document.querySelectorAll('.panel__toggle-icon')
+    .forEach((element) =>
+      element.addEventListener(
+          'click',
+          (event) => toggleIconHandler(event))
+    );
+
+// Add event listeners to pre-existing buttons in list elements
+const listElementSelector =
+    '.panel__list-unordered ' +
+    '.panel__content-entry--list ' +
+    '.panel__button-incremental--small';
+document.querySelectorAll(listElementSelector)
+    .forEach((element) =>
+      element.addEventListener(
+          'click',
+          (event) => removeTextListElement(event))
+    );
+
+/**
+ * Changes icon from clicked to un-clicked, and vice versa.
+ * Assumes parent div has two img elements, one hidden and one not.
+ *
+ * @param {Event} event event generated from click
+ */
+function toggleIconHandler(event) {
+  const currentIcon = event.target;
+  const currentDiv = currentIcon.parentNode;
+  const newIcon = currentDiv.querySelector('img[hidden]');
+
+  currentIcon.setAttribute('hidden', '');
+  newIcon.removeAttribute('hidden');
+}
+
+/**
+ * Will increase a the integer value of a text element up to a
+ * specified max.
+ *
+ * @param {string} elementId the ID of the element containing the integer value
+ * @param {number} max the maximum integer the text element should increase to.
+ *     30 by default
+ */
+function incrementElement(elementId, max = 30) {
+  const targetElement = document.getElementById(elementId);
+  const currentValue = parseInt(targetElement.innerText);
+
+  targetElement.innerText = Math.min(max, currentValue + 1);
+}
+
+/**
+ * Will decrease a the integer value of a text element down to a
+ * specified minimum.
+ *
+ * @param {string} elementId the ID of the element containing the integer value
+ * @param {number} min the minimum integer the text element should decrease to.
+ *     1 by default
+ */
+function decrementElement(elementId, min = 1) {
+  const targetElement = document.getElementById(elementId);
+  const currentValue = parseInt(targetElement.innerText);
+
+  targetElement.innerText = Math.max(min, currentValue - 1);
+}
+
+/**
+ * Add a panel__content-entry-list entry to a list with text and a
+ * remove button (to delete the entry)
+ *
+ * @param {string} inputElementId the id of the input field
+ * @param {string} listElementId the id of the list that contains the entries
+ */
+function createTextListElement(inputElementId, listElementId) {
+  const inputElement = document.getElementById(inputElementId);
+
+  // Remove commas from phrase, since this is currently the delimiter
+  // for query strings and they are not recognized by most Google services
```

Comment:
What do you mean by "not recognized by most Google services"? You can't use commas even if you escape them?

Reply:
I should clarify this in the comment. 

You **can** use them for visual purposes (i.e. naming a task with a comma, or having a tasklist that contains a comma in the title, or having a task notes with a comma). But in terms of search queries (for Gmail and otherwise), punctuation is ignored.

Reply:
ACK